new-log-viewer/src/services/MainWorker.ts

```diff
@@ -35,12 +36,31 @@ const postResp = <T extends WORKER_RESP_CODE>(
     postMessage({code, args});
 };
 
+// eslint-disable-next-line max-lines-per-function,max-statements
```

Comment:
Shall we add a TODO to break this function up (assuming we don't want to do it in this PR)?